### PR TITLE
Month filter

### PR DESCRIPTION
**Pull request by spoonben**
Adding the ability to filter by month. It's not styled yet, but take a look to see if you think it's useful.

### CODE REVIEW

File: src/components/select/months.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import * as React from "react";
+import * as R from "ramda";
+
+import Select from "react-select";
+import { monthMap } from "../constants";
+
+const getValue = (month) =>
+  Number(R.head(R.values(R.pick([month], R.invertObj(monthMap)))));
```
Comment (spoonben) [inline src/components/select/months.js:8]: I know I know I know this is gross, please give me another way and I'll take it.

File: src/search.js [hunk 2]
```diff
@@ -10,34 +10,56 @@ const critterTypes = {
...
 const sortBy = (filter) => R.sortBy(R.prop(filter));
 
-const getLeavingNow = ({ baseResults, hemisphere }) => {
-  const currentMonth = new Date().getMonth() + 1; // because javascript thinks jan is 0
-  return baseResults.filter(
-    (item) =>
-      item.available[hemisphere].includes(currentMonth) &&
-      !item.available[hemisphere].includes(currentMonth + 1)
-  );
+const filterEverything = ({
```
Comment (spoonben) [inline src/search.js:13]: I wanted to use a Ramda function for this, I still might, but this is what it is right now.

File: src/components/select/months.js (hunk 1 above)
Comment (HankMcCoy) [reply src/components/select/months.js:8]: As an alternative, we could calculate `options` using vanilla JS like this, I think?

```js
const options = Object.entries(monthMap).map(([value, label]) => ({
  value: Number(value),
  label,
}));
```

FWIW, I've soured on Ramda pretty heavily as I feel like as much as I agree with a lot of the theory behind it, whenever I see it used in practice I'm always like, "WTF is going on why does this make me feel like an idiot." 🤷🏻‍♂️2️⃣¢

File: src/search.js (hunk 2 above)
Comment (HankMcCoy) [reply src/search.js:13]: Haha, just saw this. Glad to see I'm not the only one who Ramda baffles. :)

Comment (HankMcCoy) [reply src/search.js:13]: Honestly, I think the way you have it is plenty readable. If you wanted, you could bump up the readability maybe a tad more by including a quick `//` comment above each section, e.g.:

```js
// Filter by the searched text
if (searchText) {
  ...
}

// Filter by "Leaving this month"
if (leavingNow) {
  ...
}

// Filter by month
if (!leavingNow && monthsToFilter && monthsToFilter.length > 0) {
  ...
}
```

But again, I honestly think it is fine as-is.

**Pull request merged**